### PR TITLE
SRv6 Remote Mechanism

### PR DESCRIPTION
**Pull request by artem-belov**
## Motivation and Context
#724

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [x] Added integration testing to cover
- [x] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: .mk/build.mk [hunk 1]
```diff
@@ -18,7 +18,7 @@ include $(foreach module, $(modules), ./$(module)/build.mk)
 
 BIN_DIR = $(PWD)/build/dist
 VERSION = $(shell git describe --tags --always)
-VPP_AGENT=ligato/vpp-agent:v2.3.0
+VPP_AGENT=artembelov/vpp-agent:v2.3.0
```
[the remark this answers is not in the document]
Comment (artem-belov) [reply .mk/build.mk]: I uploaded vpp-agent with more recent vpp commit with srv6 fix (no changes in vpp-agent). Next vpp-agent version will contain that vpp commit and I will return original docker hub.

File: controlplane/pkg/serviceregistry/serviceregistry.go [hunk 2]
```diff
@@ -41,6 +43,7 @@ type ServiceRegistry interface {
 	WaitForForwarderAvailable(ctx context.Context, model model.Model, timeout time.Duration) error
 
 	VniAllocator() vni.VniAllocator
+	SIDAllocator() sid.SIDAllocator
```
[the remark this answers is not in the document]
Comment (artem-belov) [reply controlplane/pkg/serviceregistry/serviceregistry.go]: We are planning to move allocators to vppagent with next refactoring

File: controlplane/pkg/nsmd/nsmd.go [hunk 3]
```diff
@@ -43,6 +43,9 @@ const (
 	NsmdDeleteLocalRegistry = "NSMD_LOCAL_REGISTRY_DELETE"
 	ForwarderTimeout        = 1 * time.Hour
 	NSEAliveTimeout         = 1 * time.Second
+
+	// PreferredRemoteMechanism - remote mechanism will be checked first
+	PreferredRemoteMechanism = "PREFERRED_REMOTE_MECHANISM"
```
Comment (haiodo) [inline controlplane/pkg/nsmd/nsmd.go]: Why we need two duplicate  constants here and controlplane/pkg/remote/forwarder_service.go

File: controlplane/pkg/remote/forwarder_service.go [hunk 4]
```diff
@@ -76,48 +81,83 @@ func (cce *forwarderService) findMechanism(mechanismPreferences []*connection.Me
...
-			continue
+	var mechanism *connection.Mechanism
+	var dpMechanism *connection.Mechanism
+
+	if preferredMechanismName := os.Getenv(PreferredRemoteMechanism); len(preferredMechanismName) > 0 {
+		for _, m := range request.GetRequestMechanismPreferences() {
+			if m.GetType() == preferredMechanismName {
+				if dpm := cce.findMechanism(dp.RemoteMechanisms, m.GetType()); dpm != nil {
+					mechanism = m
+					dpMechanism = dpm
```
Comment (haiodo) [inline controlplane/pkg/remote/forwarder_service.go]: Should we stop iteration here?

File: scripts/aws/create-kubernetes-cluster.go [hunk 5]
```diff
@@ -227,7 +225,7 @@ func (ac *AWSCluster) createEksWorkerNodes(cfClient *cloudformation.CloudFormati
 	// with Kubernetes version 1.14.7
 	// for region us-east-2.
 	// Amazon EKS-Optimized AMI list: https://docs.aws.amazon.com/eks/latest/userguide/eks-optimized-ami.html
-	eksAmi := aws.String("ami-0b031080918049726")
```
Comment (haiodo) [inline scripts/aws/create-kubernetes-cluster.go]: Could you please describe this change. It is not clear about this constant. Thanks.

File: forwarder/pkg/common/egress_interface.go [hunk 6]
```diff
@@ -36,6 +36,8 @@ type ARPEntry struct {
 // EgressInterfaceType describes the info about the egress interface used for tunneling
 type EgressInterfaceType interface {
 	SrcIPNet() *net.IPNet
+	SrcIPV6Net() *net.IPNet
+	SrcLocalSID() net.IP
```
[the remark this answers is not in the document]
Comment (artem-belov) [reply forwarder/pkg/common/egress_interface.go]: It is LocalSID property of SRv6 connection:

> rv.VppConfig.Srv6Localsids = []*vpp_srv6.LocalSID{

I have added comment to explain why it is equal local ipv6 address

File: scripts/aws/create-kubernetes-cluster.go (hunk 5 above)
Comment (artem-belov) [reply scripts/aws/create-kubernetes-cluster.go]: Just changed AWS ami image to more recent. Does not affect anything for us.
I found the way to autoselect aws ami image. Will apply in another pull request

**Pull request merged**